src/connection.rs

```diff
@@ -999,6 +1038,49 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_read_bytes_with_files() {
+        let (sender, receiver) = UnixStream::pair().unwrap();
+        receiver.set_nonblocking(true).expect("Can't modify socket");
+        let mut conn = HttpConnection::new(receiver);
+
+        // Create 2 files, edit the content and rewind back to the start.
+        let mut file1 = TempFile::new().unwrap().into_file();
+        let mut file2 = TempFile::new().unwrap().into_file();
+        file1.write(b"foo").unwrap();
+        file1.seek(SeekFrom::Start(0)).unwrap();
+        file2.write(b"bar").unwrap();
+        file2.seek(SeekFrom::Start(0)).unwrap();
+
+        // Send these 2 file descriptors along with 3 bytes of data.
+        assert_eq!(
+            sender.send_with_fds(
+                &[[1, 2, 3].as_ref()],
+                &[file1.into_raw_fd(), file2.into_raw_fd()]
+            ),
+            Ok(3)
+        );
+
+        // Check we receive the right amount of data along with the right
+        // amount of file descriptors.
+        assert_eq!(conn.read_bytes(), Ok(3));
```

Comment:
Should we also check if  `conn.buffer` contains `[1,2,3]` ?

Reply:
Agreed.